src/mtcp/mtcp_restart.c

```diff
@@ -552,6 +561,11 @@ static void restorememoryareas(RestoreInfo *rinfo_ptr)
 
   RestoreInfo restore_info;
   mtcp_memcpy(&restore_info, rinfo_ptr, sizeof (restore_info));
+  if (rinfo_ptr->saved_brk != NULL); {
```

Comment:
I am wondering if we should call mtcp_sys_brk() from here only. Why should we even bother about calling it before rinfo has been relocated?